### DISCUSSION TITLE
built-in command `source` does not evaluate variables

### DISCUSSION THREAD

**Discussion by Manuzor** [nushell/nushell#4905] [Q&A] [answered]
I'm trying to do `source $nu.config-path` on the REPL but it says the following:

```
Error: nu::parser::file_not_found (link)

  × File not found
   ╭─[entry #61:1:1]
 1 │ source $nu.config-path
   ·        ───────┬───────
   ·               ╰── File not found: $nu.config-path
   ╰────
```

Other commands such as `open $nu.config-path` or `^cat $nu.config-path` work as expected. What am I missing?

Sorry if this is something obvious, I'm new to nu.

```
〉nu --version
0.60.0
〉sys | get host | select name 'os version' 'kernel version'
╭────────────────┬───────────────────────╮
│ name           │ Windows               │
│ os version     │ 10 (22000)            │
│ kernel version │ 22000                 │
╰────────────────┴───────────────────────╯
```

**Comment by kubouch** [comment, marked as answer]
Yeah, this is by design. It can only accept string literals. You can check this section of the book that explains it: https://www.nushell.sh/book/thinking_in_nushell.html#parsing-and-evaluation-are-different-stages

**Reply by Manuzor** [reply]
I see! Thanks. I've actually read this before but I guess I didn't quite fully understand it. Is there another way to achieve this? The following works but it seems kind of silly to do it this way:

```
> echo $"source ($nu.config-path)" | save foo.nu
> source foo.nu
> rm foo.nu
```

**Reply by kubouch** [reply]
Unfortunately, not directly. However, check out LIB_DIRS: https://github.com/nushell/nushell/blob/00b67d338dd9b7274b1b11d4f84cc87a836c525c/docs/sample_config/default_config.nu#L46

LIB_DIRS environment variable lets you define a set of directories where `source` and `use` look for files, so you don't need the full absolute path, just filename. Kind of like PATH, but for nu scripts. I hope that helps.
